src/libs/actions/NetworkRequestQueue.js

```diff
@@ -1,15 +1,48 @@
 import Onyx from 'react-native-onyx';
+import _ from 'underscore';
+import lodashUnionWith from 'lodash/unionWith';
 import ONYXKEYS from '../../ONYXKEYS';
 
+const retryMap = new Map();
+let persistedRequests = [];
+
+Onyx.connect({
+    key: ONYXKEYS.NETWORK_REQUEST_QUEUE,
+    callback: val => persistedRequests = val || [],
+});
+
 function clearPersistedRequests() {
     Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, []);
+    retryMap.clear();
 }
 
 function saveRetryableRequests(retryableRequests) {
-    Onyx.merge(ONYXKEYS.NETWORK_REQUEST_QUEUE, retryableRequests);
+    persistedRequests = lodashUnionWith(persistedRequests, retryableRequests, _.isEqual);
+    Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, persistedRequests);
+}
+
+function removeRetryableRequest(request) {
+    retryMap.delete(request);
+    persistedRequests = _.reject(persistedRequests, r => _.isEqual(r, request));
+    Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, persistedRequests);
+}
+
+function incrementRetries(request) {
+    const current = retryMap.get(request) || 0;
+    const next = current + 1;
+    retryMap.set(request, next);
+
+    return next;
+}
```

Comment:
Retry counts are not persisted to disk - no need to 
If a request is retried 5 times then the app quit, when the app is launched again the request will have a fresh count of up to 10 attempts, and not only the remaining 5

Might be an edge case but I think it would be best to have a retry limit in case some corrupt data was persisted and it cannot be sent no matter how many times we try